src/main/java/seedu/address/storage/JsonAdaptedPerson.java

```diff
@@ -62,7 +63,7 @@ public JsonAdaptedPerson(Person source) {
         email = source.getEmail().value;
         address = source.getAddress().value;
         payment = source.getPayment().hasPaid.toString();
-        attendance = source.getAttendance().isPresent.toString();
+        attendance = source.getAttendance().attendanceDate.format(Attendance.VALID_DATE_FORMAT);
```

Comment:
Actually I am abit confused by this, so for the Attendance class, there is two diiferent ways to format the date, one is using toString() and one is the VALID_DATE_FORMAT, so from what I understand is that toString() is used to display the dates to user, while VALID_DATE_FORMAT is used when deserialising it from Json and checking if the user input is in a valid format. But is there a reason we are using different ways to format LocalDate? Purely just curious.

Reply:
VALID_DATE_FORMAT checks for valid user input while toString() displays the date according to another format in the UI. 

Not sure if toString() should be exactly the same format returned as the user input but it might run into some problems like the test cases that uses string to return the original user input. I tried to fix that problem using the original date and formatting using VALID_DATE_FORMAT.

Reply:
And also for storing of the attendance in this case since to string does not directly return the valid format required to instantiate a new attendance

Reply:
ahhh okayy got you.